elide-datastore/elide-datastore-hibernate5/src/main/java/com/yahoo/elide/datastores/hibernate5/HibernateTransaction.java

```diff
@@ -123,15 +120,64 @@ public void commit() {
         criterion = CriterionFilterOperation.andWithNull(criterion,
                 criterionFilterOperation.applyAll(filteredPredicates));
 
-        // if no criterion then return all objects
-        if (criterion == null) {
-            return loadObjects(loadClass);
+
+        final Pagination pagination = filterScope.hasPagination() ? filterScope.getRequestScope().getPagination()
+                : null;
+
+        // if we have sorting and sorting isn't empty, then we should pull dictionary to validate the sorting rules
+        Set<Order> validatedSortingRules = null;
+        if (filterScope.hasSortingRules()) {
+            final Sorting sorting = filterScope.getRequestScope().getSorting();
+            final EntityDictionary dictionary = filterScope.getRequestScope().getDictionary();
+            if (sorting.hasValidSortingRules(loadClass, dictionary)) {
```

Comment:
I would think sorting.hasValidSortingRules should also test for empty sorting rules (so you don't have to check again down below for empty).